api/src/main/java/org/infinispan/api/Infinispan.java

```diff
@@ -1,50 +1,100 @@
 package org.infinispan.api;
 
-import java.util.concurrent.CompletionStage;
+import java.net.URI;
+import java.util.Iterator;
+import java.util.ServiceLoader;
 
-import org.infinispan.api.configuration.ClientConfig;
-import org.infinispan.api.configuration.EmbeddedConfig;
-import org.infinispan.api.reactive.KeyValueStore;
-import org.infinispan.api.reactive.KeyValueStoreConfig;
+import org.infinispan.api.async.AsyncContainer;
+import org.infinispan.api.configuration.Configuration;
+import org.infinispan.api.exception.InfinispanConfigurationException;
+import org.infinispan.api.mutiny.MutinyContainer;
+import org.infinispan.api.sync.SyncContainer;
 
 /**
  * Infinispan instance, embedded or client, depending on the access point.
  *
- * @author Katia Aresti, karesti@redhat.com
- * @since 10.0
+ * @since 14.0
  */
 @Experimental("This is not ready yet for general consumption. Major changes are still expected.")
-public interface Infinispan {
-
-   static Infinispan newClient(ClientConfig config) {
-      try {
-         return (Infinispan) Infinispan.class.getClassLoader()
-               .loadClass("org.infinispan.api.client.impl.InfinispanClientImpl")
-               .getConstructor(ClientConfig.class)
-               .newInstance(config);
-      } catch (Exception e) {
+public interface Infinispan extends AutoCloseable {
+   /**
+    * <ul>
+    *   <li><tt>file:///.../infinispan.xml</tt> Embedded Infinispan configured via XML/JSON/Yaml file</li>
+    *   <li><tt>classpath:///.../infinispan.xml</tt> Embedded Infinispan configured via XML/JSON/Yaml classpath resource</li>
+    *   <li><tt>hotrod[s]://[username[:password]@]host:port[,host2:port]?property=value[&property=value]</tt> </li>
+    * </ul>
+    *
+    * @param uri one of the supported Infinispan URIs:
+    * @return an
+    */
+   static Infinispan create(URI uri) {
+      Iterator<Factory> factories = ServiceLoader.load(Factory.class).iterator();
```

Comment:
Is the use of TCCL intentional?  or can we instead load factory implementations from: `ServiceLoader.load(Factory.class, Factory.class.getClassLoader())` ?

Reply:
I'm now using your suggestion